### ISSUE TITLE
Analyze Phenotypes Gene Autocomplete V.S. Main search results

### ISSUE THREAD

**Issue by yuanzhou** [monarch-initiative/monarch-legacy#1287]
@harryhoch noticed this while he was testing the analyze phenotype gene search. If you type "EG" into the gene symbol autocomplete field you'll see this:

![1](https://cloud.githubusercontent.com/assets/195873/16319541/ad7799dc-3961-11e6-8c53-3a92b82bbd92.PNG)

The autocomplete results have terms that don’t string match. That might confuse end users.

And if you go to the monarch main search and enter the same "EG", you'll get this:

![2](https://cloud.githubusercontent.com/assets/195873/16319562/be7fed7e-3961-11e6-85b2-7e3b08de4102.PNG)

In this case, all autocomplete items contain the letters "EG".

**Comment by yuanzhou** [comment]
I also talked to @cborromeo and he pointed me to a couple of places regarding the differences between Analyze Phenotype gene autocomplete and the global auto search form results of the same "EG" search term. 

The main search form is using `https://scigraph-ontology-dev.monarchinitiative.org/scigraph/` to get the edges and nodes while the Analyze Phenotype gene autocomplete is using `https://scigraph-data-dev.monarchinitiative.org/scigraph/` and there are also some result filtering differences.

**Comment by harryhoch** [comment]
ok, @yuanzhou, do you know how to proceed?

**Comment by yuanzhou** [comment]
@kshefchek which scrigraph server should I use for the gene autocomplete search in Analyze Phenotypes? 

`https://scigraph-ontology-dev.monarchinitiative.org/scigraph/`

or

`https://scigraph-data-dev.monarchinitiative.org/scigraph/`

**Comment by kshefchek** [comment]
We should use the server in the global_scigraph_url variable.

**Comment by yuanzhou** [comment]
I see, currently the gene autocomplete is using `global_scigraph_data_url`.

**Comment by yuanzhou** [comment]
I think I might have fixed the gene autocomplete, https://github.com/monarch-initiative/monarch-app/pull/1286/commits/2fb70e597a3328b9828361852ea2696a1e73923a and https://github.com/monarch-initiative/monarch-app/commit/3ddbbfd48e87192dc412e4373f3876904244354a. After comparing the main autocomplete search and the gene autocomplete search I discovered that 
1. The main search sends the user-entered search term, e.g., "EG", to `/autocomplete/EG.json` and the gene search calls `/autocomplete/gene/EG.json`. Their resulting JSON files are very similar but not the same since I saw different genes.
2. The main search then filters out all the duplicates. The old gene search didn't remove equivalent gene ids.

So basically I ported some code from the main search to the gene search to remove duplicates when composing the gene id list and switched to the `global_scigraph_url`. Now the gene autocomplete search returns this:

![capture](https://cloud.githubusercontent.com/assets/195873/16657107/3af1c3b4-442f-11e6-8f97-55853e68ae13.PNG)

I'm still not 100% sure about this approach though. Please advice @harryhoch @kshefchek

**Comment by harryhoch** [comment]
@yuanzhou, this looks reasonable, but I can't comment on whether this content is correct. @kshefchek ?

**Comment by kshefchek** [comment]
In that screenshot it looks like we've lost the taxon information.

**Comment by harryhoch** [comment]
Good point, @kshefchek , @yuanzhou, any idea whats up?

**Comment by yuanzhou** [comment]
@kshefchek I'll add the species back.

**Comment by yuanzhou** [comment]
Taxon added via this commit https://github.com/monarch-initiative/monarch-app/pull/1286/commits/d2144cab2f3296303ef1a0d4a9b1b42b4320da33, and the PR will be merged soon: https://github.com/monarch-initiative/monarch-app/pull/1286

I'm closing this issue now. Will reopen it if there's any unresolved issues around this topic.
